plugins/analytics/public/github.py

```diff
@@ -0,0 +1,158 @@
+            "id": repository.id,
+            "fullname": repository.full_name,
+            "last modified": repository.last_modified,
+            "pushed at": repository.pushed_at,
+            "created at": repository.created_at,
+            "stars": repository.stargazers_count,
+            "subscribers": repository.subscribers_count,
+            "forks": repository.forks_count,
+            "watchers": repository.watchers_count,
+            "matching query": query,
+        }
+        repository_obs.add_context(self.name, context)
+        return repository_obs.save()
+
+    def create_user_observable(self, owner, tags):
+        logging.info(f"Processing owner: {owner.html_url}")
+        user_obs = self._get_or_create_observable(
+            ObservableType.user_account, owner.login
+        )
+        user_obs.user_id = str(owner.id)
+        user_obs.account_login = owner.login
+        user_obs.account_type = "github"
+        user_obs.display_name = owner.name
+        user_obs.account_created = owner.created_at
+        user_obs.tag(tags + ["github-user"])
+        user_obs.created = owner.created_at
+        context = {"bio": owner.bio, "email": owner.email, "url": owner.html_url}
+        user_obs.add_context(self.name, context)
+        return user_obs.save()
+
+    def handle_code_search(self, indicator, query, tags):
+        logging.info(f"[+] Searching code with {query}")
+        for code in self.__github_api.search_code(query):
+            code_obs = self.create_code_observable(code, query, tags)
+            repository_obs = self.create_repository_observable(
+                code.repository, query, tags
+            )
+            owner_obs = self.create_user_observable(code.repository.owner, tags)
+            owner_obs.link_to(repository_obs, "owns", "")
+            repository_obs.link_to(code_obs, "contains", f"matches {query}")
+            indicator.link_to(code_obs, "matches", f"matches {query}")
+            indicator.link_to(repository_obs, "contains", f"matched file {query}")
+
+    def handle_repositories_search(self, indicator, query, tags):
+        logging.info(f"[+] Searching repositories with {query}")
+        for repository in self.__github_api.search_repositories(query):
+            repository_obs = self.create_repository_observable(repository, query, tags)
+            owner_obs = self.create_user_observable(repository.owner, tags)
+            owner_obs.link_to(repository_obs, "owns", "")
+            indicator.link_to(repository_obs, "matches", f"matches {query}")
+
+    def run(self):
+        github_token = yeti_config.get("github", "token")
+
+        auth = Auth.Token(github_token)
+        self.__github_api = Github(auth=auth)
+
+        github_queries, _ = indicator.Query.filter({"query_type": "github"})
+        logging.info(
+            f"[+] Found {len(github_queries)} Github queries: {github_queries}"
+        )
+
+        try:
```

Comment:
More generally this will make the whole feed fail as soon as the first error is encountered and exception is raised - so potentially a single invalid query could be enough to stop the entire run of the feed.

Reply:
Do you mean here I should enclose each query call in the try rather than the whole loop?